ui/src/app/contexts/error-reporter-context.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import {createContext, useEffect, useState} from 'react';
+
+import {StackdriverErrorReporter} from 'stackdriver-errors-js';
+
+import {configApi} from 'app/services/swagger-fetch-clients';
+import {serverConfigStore as angularServerConfigStore} from 'app/utils/navigation';
+import {serverConfigStore, useStore} from 'app/utils/stores';
+import {atom} from 'app/utils/subscribable';
+import {environment} from 'environments/environment';
+
+interface StackdriverReporterStore {
+  reporter?: StackdriverErrorReporter;
+}
+
+// Exported for use in LeoRuntimeInitializer, should be removed after WorkspaceWrapper is converted
+export const stackdriverReporterStore = atom<StackdriverReporterStore>({});
+
+const StackdriverErrorReporterContext = createContext({
+  reportError: (err: (Error|string)) => {}
+});
+
+export const StackdriverReporterProvider = ({children}) => {
+  useStore(stackdriverReporterStore);
+  const [config, setConfig] = useState();
+
+  useEffect(() => {
+    if (environment.debug) {
+      // This is a local dev server, we want to disable Stackdriver reporting as
+      // it's not useful and likely won't work due to the origin.
+      return;
+    }
+
+    async function getServerConfig() {
```

Comment:
rename to `setServerConfig()`? I was confused when I saw that this function isn't returning anything

Reply:
Yeah, good point.